internal/it/skip.go

```diff
@@ -0,0 +1,327 @@
+	case skipClientVersion:
+		ensureLen(parts, 3, cond, "ver = 5.0")
+		return s.checkVer(parts[1], parts[2])
+	case skipOS:
+		ensureLen(parts, 3, cond, "os = linux")
+		return s.checkOS(parts[1], parts[2])
+	case skipArch:
+		ensureLen(parts, 3, cond, "arch = 386")
+		return s.checkArch(parts[1], parts[2])
+	case skipEnterprise:
+		ensureLen(parts, 1, cond, "enterprise")
+		return s.isEnterprise()
+	case skipNotEnterprise:
+		ensureLen(parts, 1, cond, "!enterprise")
+		return !s.isEnterprise()
+	case skipOSS:
+		ensureLen(parts, 1, cond, "oss")
+		return s.isOSS()
+	case skipNotOSS:
+		ensureLen(parts, 1, cond, "!oss")
+		return !s.isOSS()
+	default:
+		panic(fmt.Errorf(`unexpected test skip constant "%s" in %s`, parts[0], cond))
+	}
+}
+
+func ensureLen(parts []string, expected int, condition, example string) {
+	if len(parts) != expected {
+		panic(fmt.Errorf(`unexpected format for %s, example of expected condition: "%s"`, condition, example))
+	}
+}
+
+func checkVersion(left, operator, right string) bool {
+	cmp := compareVersions(left, right)
+	switch operator {
+	case "=":
+		return cmp == 0
+	case "!=":
+		return cmp != 0
+	case ">":
+		return cmp > 0
+	case ">=":
+		return cmp >= 0
+	case "<":
+		return cmp < 0
+	case "<=":
+		return cmp <= 0
+	default:
+		panic(fmt.Errorf(`unexpected test skip operator "%s" to compare versions`, operator))
+	}
+}
+
+func compareVersions(left, right string) int {
+	var leftHasSuffix, rightHasSuffix bool
+	leftHasSuffix, left = stripSuffix(left)
+	rightHasSuffix, right = stripSuffix(right)
+	// versionNumbers describe the numbers of the present version
+	leftNums := strings.Split(left, ".")
+	// checkNumbers describe the numbers received to test for
+	rightNums := strings.Split(right, ".")
+	// make rightNums and leftNums the same length by filling the shorter one with zeros.
+	if len(rightNums) < len(leftNums) {
+		rightNums = equalizeVersions(rightNums, leftNums)
+	} else if len(leftNums) < len(rightNums) {
+		leftNums = equalizeVersions(leftNums, rightNums)
+	}
+	for i := 0; i < len(rightNums); i++ {
+		r := mustAtoi(rightNums[i])
+		l := mustAtoi(leftNums[i])
+		if r < l {
+			return 1
+		}
+		if r > l {
+			return -1
+		}
+	}
+	// if the version number has a suffix, then it is prior to the non-suffixed one.
+	// See: https://semver.org/#spec-item-9
+	if leftHasSuffix {
+		if rightHasSuffix {
+			return 0
+		}
+		return -1
+	}
+	if rightHasSuffix {
+		return 1
+	}
+	return 0
+}
+
+// stripSuffix checks and removes suffix, e.g., "-beta1" from the version.
+func stripSuffix(version string) (hasSuffix bool, newVersion string) {
+	newVersion = version
+	hasSuffix = strings.Contains(version, "-")
+	if hasSuffix {
+		newVersion = strings.SplitN(newVersion, "-", 2)[0]
+	}
+	return
+}
+
+func mustAtoi(s string) int {
+	n, err := strconv.Atoi(s)
+	if err != nil {
+		panic(fmt.Errorf("could not parse int %s: %w", s, err))
+	}
+	return n
+}
+
+func checkEquality(left, operator, right, key string) bool {
```

Comment:
Just a suggestion, case-insensitive comparison can be easier to use.

Reply:
We panic if a key (like `hz`, `version`) not found, so the user cannot make a mistake. I think let's keep it so the user cannot write something like `VeRsiON < 5.1` :)